fdb-extensions/src/main/java/com/apple/foundationdb/util/Loggable.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Loggable.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.util;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ * Associates loggable information with an object as a map.
+ *
+ * @return a single map with all log information
+ */
+interface Loggable<T> {
```

Comment:
It may make sense to have `KeyValueLogMessage` implement this interface as well.

Reply:
Currently this only exposes a couple of static methods for adding a message + keys and values. If this were to implement the LoggableKeysAndValues interface it would need to expose non-static methods like getLogInfo, addLogInfo, etc.